the-collab-lab/tcl-55-smart-shopping-list
# Adds Chakra UI components for the Add Item page

## Description

This code adds Chakra UI components to the Add Item and List pages. A drawer component has a button on the List page to access the Add Item page, where it has been updated with Chakra UI form and toast components. This simplifies the code by adding Chakra UI components that enhance the design and usability of the page, while also keeping a consistent interface with the rest of the application.

## Related Issue

Closes #38 
Closes #37 

## Acceptance Criteria

- [x]  The Add Item page within the drawer includes fields for entering item details, such as name and the timeframe for buying the item again.
- [x]  Chakra UI components for input fields should have appropriate styling, including labels, placeholders, and error handling.
- [x]  The form should have a submit button, styled using Chakra UI, to allow users to submit the item details.
- [x] The error and added messages should be displayed to the user using the chakra `toast` component.
- [x]  A Chakra UI drawer component should be implemented to hold the Add Item page.
- [x]  The drawer should be accessible from the List page, allowing users to open and close it.
- [x]  When the drawer is opened, the Add Item page should be displayed within the drawer.

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
| ✓   | :sparkles: New feature     |
| ✓   | :hammer: Refactoring       |

## Updates

### Before

![image](https://github.com/the-collab-lab/tcl-55-smart-shopping-list/assets/30507348/f63e2925-00bf-429a-8291-626c7fcbc0d5)

### After

![image](https://github.com/the-collab-lab/tcl-55-smart-shopping-list/assets/30507348/bcc31c58-0b05-4a0b-a0bf-6520038558ce)

![image](https://github.com/the-collab-lab/tcl-55-smart-shopping-list/assets/30507348/07395b33-fa68-4b0e-97c9-b84709086f30)


## Testing Steps / QA Criteria

In your terminal, use `git checkout jf-tf-add-item-styling` to create a new branch and switch over.
Use `git pull origin jf-tf-add-item-styling` to bring the changes from remote to local.
npm start to start a local instance of our app

- Open the drawer from the List page and confirm that the Add Item page is displayed within it
- Test the drawer accessibility by opening and closing it using both keyboard and mouse
- Add an item to the list and verify that it appears on the list page
- Test error handling by entering invalid data and confirm that appropriate error messages are displayed using the Chakra toast component
